### DISCUSSION TITLE
about supported pprint functions

### DISCUSSION THREAD

**Discussion by kangaroolab** [borkdude/scittle#86] [General]
is it correct only following functions available from pprint plugin?

cl-format
pprint
print-table

**Comment by borkdude** [comment]
Hi @kangaroolab, this is correct:

https://github.com/babashka/sci.configs/blob/33575ca723f54af3e5bd5a643e51b480a157054b/src/sci/configs/cljs/pprint.cljs#L22

Feel free to request more stuff, we can add it in sci.configs

**Comment by kangaroolab** [comment]
thanks...could we possibly include following functions in the plugin with scittle?

(p/**with-pprint-dispatch**
  p/**code-dispatch**
  (p/pprint 
    (clojure.edn/read-string lost-formatting))))

**Reply by borkdude** [reply]
Done in sci.configs. If you can test this in the playground: https://babashka.org/sci.configs/
then I'll bump the configuration in scittle (and nbb).

**Comment by kangaroolab** [comment]
thanks...I ran a few tests and it worked as expected. can you please merge and share link for updated js file?

**Reply by borkdude** [reply]
done, v0.6.19

**Reply by borkdude** [reply]
Links:

https://cdn.jsdelivr.net/npm/scittle@0.6.19/dist/scittle.js
https://cdn.jsdelivr.net/npm/scittle@0.6.19/dist/scittle.cljs-ajax.js
https://cdn.jsdelivr.net/npm/scittle@0.6.19/dist/scittle.reagent.js
https://cdn.jsdelivr.net/npm/scittle@0.6.19/dist/scittle.re-frame.js
https://cdn.jsdelivr.net/npm/scittle@0.6.19/dist/scittle.promesa.js
https://cdn.jsdelivr.net/npm/scittle@0.6.19/dist/scittle.pprint.js
https://cdn.jsdelivr.net/npm/scittle@0.6.19/dist/scittle.nrepl.js

**Comment by kangaroolab** [comment]
thanks but I got following error with scittle and pprint of 0.6.19 version. no issue if I revert back to 0.6.17
```
Uncaught ReferenceError: $APP is not definedUnderstand this error
cdn.jsdelivr.net/npm…ist/scittle.js:1472 Uncaught TypeError: b.call is not a functionUnderstand this error
VM26 scittle.pprint.js:107 Uncaught ReferenceError: $APP is not definedUnderstand this error
VM67 scittle.js:1472 Uncaught TypeError: b.call is not a function
    at $APP.m.g (VM67 scittle.js:1472:348)
    at pia (VM68 scittle.pprint.js:75:104)
    at ria (VM68 scittle.pprint.js:78:270)
    at Function.<anonymous> (VM68 scittle.pprint.js:86:47)
    at eg (VM67 scittle.js:90:327)
    at $APP.Fg.h (VM67 scittle.js:784:140)
    at qZ (VM68 scittle.pprint.js:3:479)
    at Function.S_ (VM68 scittle.pprint.js:85:307)
    at eg (VM67 scittle.js:90:327)
    at $APP.Fg.h (VM67 scittle.js:784:140)
    at d (VM67 scittle.js:158:305)
    at c (VM67 scittle.js:158:221)
    at VM68 scittle.pprint.js:227:168
    at VM68 scittle.pprint.js:227:497
    at VM68 scittle.pprint.js:256:4
$APP.m.g @ VM67 scittle.js:1472
pia @ VM68 scittle.pprint.js:75
ria @ VM68 scittle.pprint.js:78
(anonymous) @ VM68 scittle.pprint.js:86
eg @ VM67 scittle.js:90
$APP.Fg.h @ VM67 scittle.js:784
qZ @ VM68 scittle.pprint.js:3
S_ @ VM68 scittle.pprint.js:85
eg @ VM67 scittle.js:90
$APP.Fg.h @ VM67 scittle.js:784
d @ VM67 scittle.js:158
c @ VM67 scittle.js:158
(anonymous) @ VM68 scittle.pprint.js:227
(anonymous) @ VM68 scittle.pprint.js:227
(anonymous) @ VM68 scittle.pprint.js:256Understand this error
```

**Reply by borkdude** [reply]
Note that you must always upgrade all the scittle files to the newest version. You can't mix version `0.6.19` with `0.6.17` for example.

I can't reproduce this with the following HTML.

``` html
<html>
  <head>
    <script src="https://cdn.jsdelivr.net/npm/scittle@0.6.19/dist/scittle.js" type="application/javascript"></script>
    <script src="https://cdn.jsdelivr.net/npm/scittle@0.6.19/dist/scittle.pprint.js" type="application/javascript"></script>
    <script type="application/x-scittle">
      (require '[cljs.pprint :as pp])
      (set! (.-innerText js/app) (with-out-str (pp/with-pprint-dispatch pp/code-dispatch (pp/pprint {:a 1}))))
    </script>
  </head>
  <body>
    <pre id="app"></pre>
  </body>
</html>
```

Feel free to post a repro in case you still have a problem.

**Comment by kangaroolab** [comment]
thanks. it works.

I tested with flems.io, where I got the error. if you try your example above, it seems keeping loading...not sure why.

**Reply by borkdude** [reply]
Ok, let's not deal with third party software. I assume that the issue is solved now.
